woocommerce/woocommerce-gutenberg-products-block
# Product Collection - Add Inherit query from template control

> ⚠️ This PR will be merged after PR #9634, as It also contains changes made in #9634

This PR adds a setting to Product Collection to inherit the global query.
<img width="270" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/815a4676-5f2e-4819-a251-92b5a4d1049f">


When enabled:
- all the filters and ordering options are hidden
- the actual global query should be in use
- make sure Product Collection works fine with the Catalog Sorting and Product Results Count blocks

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9358

### Testing

#### Testing with Product Archive templates

1. Open the Editor and navigate to the "Product Search Results"
2. Replace the existing 'Classic Template' block with the 'Product Collection' block.
3. After the 'Product Collection' block is successfully inserted, ensure that the 'Inherit query from template' option is enabled by default.
4. Save the changes.
5. Navigate to the frontend of your website. You can emulate a search for "shirt" by visiting the URL "/shop?s=shirt". Confirm that the search query `shirt` effectively displays related products.
6. Head back to the Editor.
7. In the 'Product Collection' block, deactivate the 'Inherit query from template' option.
8. Subsequently, you should observe various settings and filters available in the inspector controls. Modify these filters and settings as required and validate immediately that the changes are visible within the Editor.
9. Save your modifications and return to the frontend of your website.
10. Ensure that the adjustments made to the 'Product Collection' block within the Editor are accurately displayed on the frontend of your website.

Here is a quick demo:

https://github.com/woocommerce/woocommerce-blocks/assets/16707866/5703fa10-897c-4310-8ad1-f9c176014737


#### Testing in Post

1. Create a new post
2. Add Product Collection block to the post
3. With the 'Product Collection' block selected, open the Inspector Controls (located in the right sidebar). Confirm that the option `inherit query from template` does not appear in the list of options. This is because it's relevant only for Product Archive pages and should not be visible in this context.

https://github.com/woocommerce/woocommerce-blocks/assets/16707866/b4dce46a-7853-42f4-ae40-545850d121e9



### WooCommerce Visibility




### Changelog

> Product Collection - Add `Inherit query from template` control
